test/old_tests/UnitTests/References.cpp

```diff
@@ -27,4 +27,18 @@ TEST_CASE("References")
 
     // Tests short circuit when one is empty.
     REQUIRE(a != e);
+
+    // Tests comparison to value.
+    REQUIRE(a == 123);
+    REQUIRE(123 == a);
+
+    REQUIRE(a != 124);
+    REQUIRE(124 != a);
+
+    // Tests short circuit when reference is empty.
+    REQUIRE_FALSE(e == 123);
+    REQUIRE_FALSE(123 == e);
+
+    REQUIRE(e != 123);
+    REQUIRE(123 != e);
```

Comment:
Can you add tests for strings? They have a way of weeding out issues.

Reply:
Trying to use strings will "box" here because the raw string will be converted to `hstring` beforehand (because of `operator==(IReference<hstring>, hstring)`)... Maybe add `param::hstring` overloads and independent tests for those cases?

Reply:
Yep, it gets expensive real quick. `param::hstring` should help. Look at how `box_value` is overloaded for strings.

Reply:
Another idea would be to make the non-reference parameter generic and letting `hstring`'s == overload handle it:

```cpp
template <typename T, typename U>
bool operator==(IReference<T> const& left, U const& right)
{
    if (!left)
    {
        return false;
    }

    return left.Value() == right;
}
```

Reply:
It's worth a try. 😉

Reply:
After trying it seems that this breaks comparing an IReference with nullptr so I'll add hstring overloads